## Task #1

In [54]:
ds = (200, 800)
m_x = lambda x: sum(x) / len(x)
m_x(ds)

500.0

In [55]:
d_x = lambda x: (x[-1]-x[0])**2 / 12
d_x(ds)

30000.0

## Task #2

In [70]:
from numpy import sqrt

In [71]:
d = 0.2
a = 0.5
b_1 = sqrt(12*d) + a
print(f'b #1 = {b_1}')

b #1 = 2.049193338482967


In [72]:
print(f'mean = {m_x((a,b_1))}')

mean = 1.2745966692414834


In [73]:
b_2 = - sqrt(12*d) + a
b_2 # корень не походит, так как b > a
print(f'b #2 = {b_2}')
print(f'But "b" must be greater than "a"')

b #2 = -1.0491933384829668
But "b" must be greater than "a"


In [74]:
print(f'Res: b = {b_1}, mean = {m_x((a,b_1))}')

Res: b = 2.049193338482967, mean = 1.2745966692414834


## Task #3

$$f(x) = \frac{1}{\sigma \sqrt{2 \pi}} e ^ {- \frac{(x - a)^2}{2 \sigma^2}},$$
$$$$
$$f(x) = \frac{1}{4\sqrt{2\pi}} \cdot e^{-\frac{(x+2)^2}{32}}$$

$\sigma = 4$ - среднее квадратическое отклонение

$a = M(X) = - 2$

$D = \sigma^2 = 16$

## Task #4

$M(X) = 174$

$\sigma = 8$

$$f(x) = \frac{1}{8\sqrt{2\pi}} \cdot e^{-\frac{(x-174)^2}{128}}$$

а) Рост больше 182 см

In [83]:
sigma = 8
a = 174
z_f = lambda val: (val - a)/sigma
z = z_f(182)
z

1.0

$z = 1$ - смотрим таблицу накопленного нормального распределения

In [88]:
1 - 0.8413

0.15869999999999995

$f(x) = 1 - 0.8413 = 0.1587$

б) Рост больше 190 см

In [84]:
z = z_f(190)
z

2.0

In [85]:
1 - 0.9772

0.022800000000000042

$f(x) = 1 - 0.9772 = 0.0228$

в) Рост от 166 до 190 см

Решением будет разница вероятностей между $x = 190$ и $x = 166$

In [89]:
print(z_f(190), z_f(166))

2.0 -1.0


In [90]:
0.9772 - 0.1587

0.8185

$f(x) = 0.8185$

г) Рост от 166 до 182 см

Решением будет разница вероятностей между $x = 182$ и $x = 166$

In [95]:
print(z_f(182), z_f(166))

1.0 -1.0


In [91]:
0.8413 - 0.1587

0.6826000000000001

$f(x) = 0.6826$

д) Рост от 158 до 190 см

In [93]:
print(z_f(190), z_f(158))

2.0 -2.0


In [94]:
0.9772 - 0.0228

0.9543999999999999

$f(x) = 0.9544$

е) Рост не выше 150 или не ниже 190 см

In [96]:
print(z_f(190), z_f(150))

2.0 -3.0


In [97]:
(1 - 0.9772) + 0.0014

0.02420000000000004

$f(x) = 0.0242$

ё) Рост не выше 150 или не ниже 198 см

In [98]:
print(z_f(198), z_f(150))

3.0 -3.0


In [99]:
(1 - 0.9986) + 0.0014

0.002799999999999957

$f(x) = 0.0028$

ж) Рост ниже 166 см

In [100]:
print(z_f(166))

-1.0


In [101]:
0.1587

0.1587

$f(x) = 0.1587$

## Task #5

$M(X) = 178 см$

$D(X) = 25 см^2$

$\sigma = \sqrt{D(X)}=5$

In [103]:
z_f = lambda a, sigma, val: (val - a)/sigma
z_f(178, 5, 190)

2.4

Ответ: на 2.4 сигмы